src/box/tuple_format.c

```diff
@@ -1532,7 +1564,16 @@ tuple_format_apply_defaults(struct tuple_format *format, const char **data,
 	 */
 	for ( ; i < format->default_field_count; i++) {
 		struct tuple_field *field = tuple_format_field(format, i);
-		if (tuple_field_has_default(field)) {
+		if (tuple_field_has_default_func(field)) {
+			struct field_default_value *d = &field->default_value;
+			const char *ret_data;
+			uint32_t ret_size;
+			if (d->func->call(d->func, d->data, d->size,
+					  &ret_data, &ret_size) != 0)
+				return -1;
+			tuple_builder_add(&builder, ret_data, ret_size, 1);
+			is_tuple_changed = true;
+		} else if (tuple_field_has_default_static(field)) {
```

Comment:
Nit: I wouldn't introduce these helper functions (`has_default_func`, `has_default_static`) and instead access the fields directly here because you access them anyway right below. The code would look more straightforward then.

Reply:
Removed `tuple_field_has_default_static`, but left `tuple_field_has_default_func`, because now it compares the function id, I think it's easier to read.